### PR TITLE
Add TransportCatalog.

### PR DESCRIPTION
**Pull request by siarheishut**
Add database class TransportCatalog and unit-tests for it. The class is going to be the one which maintains data and shares with other classes.

### CODE REVIEW

File: src/distance_computer.cpp [hunk 1]
```diff
@@ -30,4 +30,29 @@ double ComputeRoadDistance(const std::vector<std::string> &stops,
   }
   return distance;
 }
+
+double ComputeGeoDistance(const std::vector<std::string> &stops,
+                          const utils::Stops &dict) {
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply src/distance_computer.cpp:35]: Done.

File: src/transport_catalog.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+  for (auto &stop : stops)
+    stops_[stop].buses.push_back(*bus_it);
+
+  info.stops = {std::make_move_iterator(stops.begin()),
+                std::make_move_iterator(stops.end())};
+
+  info.endpoints = std::move(endpoints);
+}
+}
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply src/transport_catalog.cpp:148]: Done.

File: src/transport_catalog.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#include "request_types.h"
+#include "sphere.h"
+
+namespace rm {
+class TransportCatalog {
+ public:
+  static std::unique_ptr<TransportCatalog> Create(
+      std::vector<utils::PostRequest> requests);
+
+  const utils::Stops &GetStops() const;
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply src/transport_catalog.h:21]: Done.

File: src/transport_catalog.cpp [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+namespace {
+// Latitude value range is [-90, 90];
+// Longitude value range is [-180, 180].
+// We can read about it in the following article:
+// https://www.techtarget.com/whatis/definition/latitude-and-longitude.
+const double kMinLatitude = -90.0, kMaxLatitude = 90.0,
+    kMinLongitude = -180.0, kMaxLongitude = 180.0;
+
+int ComputeUniqueCount(std::vector<std::string> stops) {
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply src/transport_catalog.cpp:28]: Done.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**